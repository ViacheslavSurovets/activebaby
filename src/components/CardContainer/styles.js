import styled from 'styled-components';
import { mobile, tablet } from '@utils/media';

export const CardWrapper = styled.section`
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin: 2vw 3vw;
  position:relative;
  ${ tablet }{
    flex-direction: column;
    .js-linkCardHover:nth-child(2){
      margin-top: 1rem;
    }
  }
  ${ mobile }{
    flex-direction: column;
    .js-linkCardHover:nth-child(2){
      margin-top: 1rem;
    }
  }
`;