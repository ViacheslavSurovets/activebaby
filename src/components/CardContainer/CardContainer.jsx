import React from 'react';
import { LinkCardHover } from '@components';
import { CardWrapper } from './styles';
import { connect } from 'react-redux';
import { selectCardData } from '@redux/homeData/homeData.selectors';
import { createStructuredSelector } from 'reselect';
import PropTypes from 'prop-types';


const CardContainer = ( { cardData } ) => {

  return(
    <CardWrapper>
      { cardData.map ( ( { id, ...rest } ) =>
        <LinkCardHover key={ id } data={ rest } />
      ) }
    </CardWrapper>
  );
};

const mapStateToProps = createStructuredSelector ( {
  cardData: selectCardData
} );

export default connect ( mapStateToProps ) ( CardContainer );

CardContainer.propTypes = {
  cardData: PropTypes.array
};
