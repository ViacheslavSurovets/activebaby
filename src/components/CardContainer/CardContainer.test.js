import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import CardContainer from './CardContainer';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('CardContainer', ()=> {
  it('CardContainer Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <CardContainer  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
