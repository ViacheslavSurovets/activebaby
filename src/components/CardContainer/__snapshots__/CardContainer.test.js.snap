// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer CardContainer Snapshot 1`] = `ShallowWrapper {}`;
